.github/workflows/clippy.yaml

```diff
@@ -0,0 +1,22 @@
+name: Clippy
+
+on:
+  pull_request:
+    branches: [ main ]
+  push:
+    branches: [ main ]
+
+
+jobs:
+  lints:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Rust toolchain and cache
+        uses: actions-rust-lang/setup-rust-toolchain@v1.5.0
```

Comment:
If this is added, it should use the same stuff for managing the toolchain as what's in the other CI jobs.

Reply:
Yes, definitely. This needs to use the same Rust setup tooling as the other workflows. If there is a reason to change that (which their might be but that should be weighed in a different issue/PR) we should change all of them to be consistent.

Reply:
changed it to use the same action as the other workflows